kedro/extras/datasets/redis/redis_dataset.py

```diff
@@ -0,0 +1,186 @@
+                pickle.loads: https://docs.python.org/3/library/pickle.html#pickle.loads
+                dill.loads: https://dill.readthedocs.io/en/latest/dill.html#dill._dill.loads
+                compress_pickle.loads:
+                https://lucianopaz.github.io/compress_pickle/html/api/compress_pickle.html#compress_pickle.compress_pickle.loads
+                All defaults are preserved.
+            save_args: Pickle options for saving pickle files.
+                You can pass in arguments that the backend dump function specified accepts, e.g:
+                pickle.dumps: https://docs.python.org/3/library/pickle.html#pickle.dump
+                dill.dumps: https://dill.readthedocs.io/en/latest/dill.html#dill._dill.dumps
+                compress_pickle.dumps:
+                https://lucianopaz.github.io/compress_pickle/html/api/compress_pickle.html#compress_pickle.compress_pickle.dumps
+                All defaults are preserved.
+            credentials: Credentials required to get access to the redis server.
+                E.g. `{"password": None}`.
+            redis_args: Extra arguments to pass into the redis client constructor
+                ``redis.StrictRedis.from_url``. (e.g. `{"socket_timeout": 10}`), as well as to pass
+                to the ``redis.StrictRedis.set`` through nested keys `from_url_args` and `set_args`.
+                Here you can find all available arguments for `from_url`:
+                https://redis-py.readthedocs.io/en/stable/connections.html?highlight=from_url#redis.Redis.from_url
+                All defaults are preserved, except `url`, which is set to `redis://127.0.0.1:6379`.
+                You could also specify the url through the env variable ``REDIS_URL``.
+
+        Raises:
+            ValueError: If ``backend`` does not satisfy the `pickle` interface.
+            ImportError: If the ``backend`` module could not be imported.
+        """
+        super().__init__()
+        try:
+            imported_backend = importlib.import_module(backend)
+        except ImportError as exc:
+            raise ImportError(
+                f"Selected backend '{backend}' could not be imported. "
+                "Make sure it is installed and importable."
+            ) from exc
+
+        if not (
+            hasattr(imported_backend, "loads") and hasattr(imported_backend, "dumps")
+        ):
+            raise ValueError(
+                f"Selected backend '{backend}' should satisfy the pickle interface. "
+                "Missing one of `loads` and `dumps` on the backend."
+            )
+
+        self._backend = backend
+
+        self._key = key
+
+        _redis_args = deepcopy(redis_args) or {}
+        self._redis_from_url_args = _redis_args.pop("from_url_args", {})
+        self._redis_from_url_args.setdefault("url", self.DEFAULT_REDIS_URL)
+        self._redis_set_args = _redis_args.pop("set_args", {})
```

Comment:
These kinda feel like `save_args` to me as well, but I get that you wanted to group redis arguments together.